components/connector/internal/certificates/loader.go

```diff
@@ -16,42 +18,93 @@ type Loader interface {
 
 type certLoader struct {
 	certificatesCache           Cache
+	readinessCh                 chan<- struct{}
+	apiServerClient             discovery.ServerVersionInterface
 	repository                  secrets.Repository
 	caSecretName                types.NamespacedName
 	rootCACertificateSecretName types.NamespacedName
+	exitCh                      <-chan struct{}
 }
 
 func NewCertificateLoader(certificatesCache Cache,
 	repository secrets.Repository,
 	caSecretName types.NamespacedName,
-	rootCACertificateSecretName types.NamespacedName) Loader {
+	rootCACertificateSecretName types.NamespacedName,
+	readinessCh chan<- struct{}, apiServerClient discovery.ServerVersionInterface,
+	exitCh <-chan struct{}) Loader {
 	return &certLoader{
 		certificatesCache:           certificatesCache,
+		readinessCh:                 readinessCh,
+		apiServerClient:             apiServerClient,
 		repository:                  repository,
 		caSecretName:                caSecretName,
 		rootCACertificateSecretName: rootCACertificateSecretName,
+		exitCh:                      exitCh,
 	}
 }
 
 func (cl *certLoader) Run() {
+	notificationCh := make(chan struct{}, 1)
+	shouldNotify := true
+	go cl.notifyOnConnection(notificationCh)
+
 	for {
-		if cl.caSecretName.Name != "" {
-			cl.loadSecretToCache(cl.caSecretName)
+		allLoaded := cl.loadSecretsToCache()
+		if allLoaded && shouldNotify {
+			shouldNotify = false
+			log.Info("All needed secrets are loaded, notifying readiness")
+			cl.readinessCh <- struct{}{}
```

Comment:
We can consider closing this channel because after we write in it once, it is not reachable anymore.